deepchecks/vision/checks/distribution/new_labels.py

```diff
@@ -0,0 +1,204 @@
+                if label not in self._class_id_counter:
+                    self._class_id_counter[label] = [0, 0]
+                self._class_id_counter[label][dataset_kind == DatasetKind.TEST] += 1
+
+    def compute(self, context: Context) -> CheckResult:
+        """Calculate which class_id are only available in the test data set and display them.
+
+        Returns
+        -------
+        CheckResult
+            value: A dictionary showing new class_ids introduced in the test set and number of times they were spotted.
+            display: Images containing said class_ids from the test set.
+        """
+        data = context.get_data_by_kind(DatasetKind.TEST)
+        class_id_only_in_test = defaultdict(list)
+        for class_id, value in self._class_id_counter.items():
+            if value[0] == 0:
+                class_id_only_in_test[class_id] = value[1]
+        class_id_only_in_test = dict(sorted(class_id_only_in_test.items(), key=lambda item: -item[1]))
+
+        # Create display
+        display = []
+        for class_id, num_occurrences in class_id_only_in_test.items():
+            # Create id of alphabetic characters
+            sid = ''.join([choice(string.ascii_uppercase) for _ in range(3)])
+            images_of_class_id = list(set(data.classes_indices[class_id]))[:self.max_images_to_display_per_label]
+            images_combine = ''.join([f'<div class="{sid}-item">{draw_image(data, x, class_id)}</div>'
+                                      for x in images_of_class_id])
+
+            html = HTML_TEMPLATE.format(
+                label_name=data.label_id_to_name(class_id),
+                images=images_combine,
+                count=format_number(num_occurrences),
+                id=sid
+            )
+            display.append(html)
+        class_id_only_in_test['all_labels'] = sum(data.n_of_samples_per_class.values())
+        return CheckResult(class_id_only_in_test, display=''.join(display[:self.max_new_labels_to_display]))
+
+    def add_condition_new_label_percentage_not_greater_than(self, max_allowed_new_labels: float = 0.005):
+        # Default value set to 0.005 because of sampling mechanism
+        """
+        Add condition - Percentage of labels that apper only in the test set required to be below specified threshold.
+
+        Parameters
+        ----------
+        max_allowed_new_labels: float , default: 0.005
+            the max threshold for percentage of labels that only apper in the test set.
+        """
+
+        def condition(result: Dict) -> ConditionResult:
+            total_labels_in_test_set = result['all_labels']
+            new_labels_in_test_set = sum(result.values()) - total_labels_in_test_set
+            present_new_labels = new_labels_in_test_set / total_labels_in_test_set
+
+            if present_new_labels > max_allowed_new_labels:
+                massage = f'{format_percent(present_new_labels)} of labels found in test set were not in train set.'
```

Comment:
Would be better if the fail message would list the class names of the top new labels.

Reply:
I think fail massage should be concise but I will add some info on top classes